types/polygon/index.d.ts

```diff
@@ -0,0 +1,210 @@
+// Type definitions for polygon 1.0
+// Project: https://github.com/tmpvar/polygon.js#readme
+// Definitions by: Konrad Klockgether <https://github.com/Nielio>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/* tslint:disable:array-type */  // cause contradictory error messages
+
+import Vec2 from 'vec2';
+
+/**
+ * Create a new polygon:
+ *
+ * ```javascript
+ * var p = new Polygon([
+ * Vec2(0, 0),
+ * Vec2(10, 0),
+ * Vec2(0, 10)
+ * ]);
+ *
+ * ```
+ *
+ * You can pass an array of `Vec2`s, arrays `[x, y]`, or objects `{ x: 10, y: 20 }`
+ */
+declare class Polygon {
+    points: Vec2[];
+
+    /**
+     * Returns the number of points in this polygon
+     */
+    readonly length: number;
+
+    constructor(points: Vec2[] | number[][] | { x: number, y: number }[]);
+
+    /**
+     * Something like Array.forEach on points
+     */
+    each(fn: (prev: Vec2, current: Vec2, next: Vec2, idx: number) => any): Polygon;
+
+    /**
+     * Returns the point at index `idx`. note: this will wrap in both directions
+     * @param idx
+     */
+    point(idx: number): Vec2;
+
+    /**
+     * Ensure all of the points are unique
+     * @param returnNew If passed a truthy value, a new Polygon will be returned to you. Otherwise the operation will be applied to this and this will be returned.
+     */
+    dedupe(returnNew?: boolean): Polygon;
+
+    /**
+     * Insert `vec2` at the specified index
+     * @param vec2
```

Comment:
Nit: You can ditch all the doc tags with no descriptions.

Reply:
I dropped them